core/sql/generator/GenExplain.cpp

```diff
@@ -660,6 +660,269 @@ FileScan::addSpecificExplainInfo(ExplainTupleMaster *explainTuple,
+				 char * currPtr2 = (char*)pushedDownColumns->getCurr();
+			      char * colNamePtr1 = NULL;
+				  Lng32 currPos = 0;
+				  short colNameLen = *(short*)currPtr2;
+				  currPos += sizeof(short);
+				  char colName[500];
+				  for (Lng32 i = 0; i < colNameLen; i++)
+				    {
+				      colName[i] = currPtr2[currPos];
+				      currPos++;
+				    }
+				  colName[colNameLen] = 0;
+				  colNamePtr1 = colName;
+				  str_sprintf(buf2, "%s",colNamePtr1);
+			}
+			pushedDownColumns->advance();
+		}
+
+
+		char* colNamePtr = buf2;
+		if(strcmp(currPtr,"EQUAL")==0){
+			 str_sprintf(buf, "(%s=?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"NOT_EQUAL")==0){
+			 str_sprintf(buf, "(%s!=?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"LESS")==0){
+			 str_sprintf(buf, "(%s<?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if(strcmp(currPtr,"LESS_OR_EQUAL")==0){
+			 str_sprintf(buf, "(%s<=?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"GREATER")==0){
+			 str_sprintf(buf, "(%s>?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"GREATER_OR_EQUAL")==0){
+			 str_sprintf(buf, "(%s>=?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"NO_OP")==0){//should never happen
+			 str_sprintf(buf, "(%s??)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"EQUAL_NULL")==0){
+			 str_sprintf(buf, "(%s=.?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"NOT_EQUAL_NULL")==0){
+			 str_sprintf(buf, "(%s!=.?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"LESS_NULL")==0){
+			 str_sprintf(buf, "(%s<.?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"LESS_OR_EQUAL_NULL")==0){
+			 str_sprintf(buf, "(%s<=.?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"GREATER_NULL")==0){
+			 str_sprintf(buf, "(%s>.?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"GREATER_OR_EQUAL_NULL")==0){
+			 str_sprintf(buf, "(%s>=.?)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"NO_OP_NULL")==0){
+			 str_sprintf(buf, "(%s?.?)",colNamePtr);//should never happen
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"IS_NULL")==0){
+			 str_sprintf(buf, "(%s is_null)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"IS_NULL_NULL")==0){
+			 str_sprintf(buf, "(%s is_null.)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"IS_NOT_NULL")==0){
+			 str_sprintf(buf, "(%s is_not_null)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"IS_NOT_NULL_NULL")==0){
+			 str_sprintf(buf, "(%s is_not_null.)",colNamePtr);
+			 outNAString += buf;
+		 }
+		 else if (strcmp(currPtr,"AND")==0)
+			  outNAString += "AND";
+		 else if (strcmp(currPtr,"OR")==0)
+			  outNAString += "OR";
+
```

Comment:
that would be a bug in the code. There should never be any not all above. So this bug will translate into not displaying any new operator that we failed to add in the display code.